### ISSUE TITLE
Implement Shape

### ISSUE THREAD

**Issue by jsbean** [dn-m/GraphicsTools#29]
```Swift
struct Shape {
    
    let path: Path

    struct Fill {

        enum Rule {
            case nonZero, evenOdd
        }
 
        let rule: Rule
        let color: Color
    }

    struct Stroke {
 
        struct Dashes {
            let pattern: [Double]
            let phase: Double
        }

        enum Cap {
            case round, butt, square
        }

        enum Join {
            case miter(limit: Double) 
            case round
            case bevel
        } 
    
        let color: Color
        let width: Double
        let cap: Cap
        let join: Join   
        let dashes: Dashes
    }
}
```

**Comment by jsbean** [comment]
Add initializers to/from `CAShapeLayer`.

**Comment by jsbean** [comment]
Closed by 2e81220c1d0614cc9329134c0063bc9f2b53770c.
